sdk/storage/azure-storage-blob/azure/storage/blob/aio/container_client_async.py

```diff
@@ -764,6 +767,119 @@ def walk_blobs(
             timeout=timeout,
             **kwargs)
 
+    @distributed_trace_async
+    async def delete_blobs(
+            self, *blobs,  # type: Union[str, BlobProperties]
+            delete_snapshots=None,  # type: Optional[str]
+            lease=None,  # type: Optional[Union[str, LeaseClient]]
+            timeout=None,  # type: Optional[int]
+            **kwargs
+        ):
+        # type: (...) -> None
+        """Marks the specified blobs or snapshots for deletion.
+
+        The blob is later deleted during garbage collection.
+        Note that in order to delete a blob, you must delete all of its
+        snapshots. You can delete both at the same time with the Delete
+        Blob operation.
+
+        If a delete retention policy is enabled for the service, then this operation soft deletes the blob or snapshot
+        and retains the blob or snapshot for specified number of days.
+        After specified number of days, blob's data is removed from the service during garbage collection.
+        Soft deleted blob or snapshot is accessible through List Blobs API specifying `include="deleted"` option.
+        Soft-deleted blob or snapshot can be restored using Undelete API.
+
+        :param blob: The blob with which to interact. If specified, this value will override
+         a blob value specified in the blob URL.
+        :type blob: str or ~azure.storage.blob.models.BlobProperties
+        :param str delete_snapshots:
```

Comment:
How about adding DeleteSnapshotsOptionType under azure.storage.blob._generated.models as an optional type?

Reply:
This PR is really about batch, so changing "delete_blob" API itself it out of scope. If you believe that's a good idea, let's discuss with arch board :)

My two cents on this is we should not expose `_generated`, since it's not a public package.